pkg/autoscaler/metrics/stats_scraper.go

```diff
@@ -181,7 +189,111 @@ func (s *serviceScraper) Scrape(window time.Duration) (Stat, error) {
 	if readyPodsCount == 0 {
 		return emptyStat, nil
 	}
-	return s.scrapeService(window, readyPodsCount)
+
+	startTime := time.Now()
+	scrapeTime := time.Since(startTime)
+	pkgmetrics.RecordBatch(s.statsCtx, scrapeTimeM.M(float64(scrapeTime.Milliseconds())))
```

Comment:
Should these be defered?

Reply:
Yeah they should. Thought I did
:facepalm: